lib/mayaHydra/hydraExtensions/sceneIndex/mayaHydraSceneIndex.cpp

```diff
@@ -1424,6 +1422,24 @@ void MayaHydraSceneIndex::_RemoveRenderItem(const MayaHydraRenderItemAdapterPtr&
     _renderItemsAdapters.erase(primPath);
 }
 
+void MayaHydraSceneIndex::GetLightedPrimPaths(SdfPathVector& lightedPrimPaths)
+{
+    _MapAdapter<MayaHydraRenderItemAdapter>(
+        [&](MayaHydraRenderItemAdapter* a) {
+            if (a->GetPrimitive() == MHWRender::MGeometry::Primitive::kTriangles) {
+                lightedPrimPaths.emplace_back(a->GetID());
+            }
+        },
+        _renderItemsAdapters);
+    _MapAdapter<MayaHydraDagAdapter>(
+        [&](MayaHydraDagAdapter* a) {
+            if (a->HasType(HdPrimTypeTokens->mesh)) {
```

Comment:
I think this is too restrictive, we should include all Renderable Prims with something like :
// This includes implicit primitive types.
static const VtArray<TfToken> GEOMETRY_PRIM_TYPES = {
    HdRprimTypeTokens->allTokens.begin(),
    HdRprimTypeTokens->allTokens.end()
};

For example, volumes should receive the lighting, I think also curves and points if they support it.
I would put this in a function so we can update it if we want like 
 const VtArray<TfToken> GetLightedPrimitivesType()

Reply:
Good idea, I'll try it